### DISCUSSION TITLE
Binding a method to an object property change event

### DISCUSSION THREAD

**Discussion by Ilnazz** [reactiveui/ReactiveUI#3697] [Q&A]
Have a nice day and thank you for the library you are developing.
Some time ago, for the needs of my own project, I developed a library for performing MVVM-like data bindings - [EasyBindings](https://github.com/Ilnazz/EasyBindings).
Here is an example of binding a method to an object property change event:
```c#
using CommunityToolkit.Mvvm.ComponentModel;
using EasyBindings;

// Simple observable object
public partial class Person : ObservableObject
{
    [ObservableProperty]
    private string _name;
}

public class Program
{
    public void Do()
    {
        var person = new Person();

        // Binding a method to a property change event
        TriggerBinder.OnPropertyChanged(context: this, observable: person, x => x.Name, newName =>
        {
            Console.WriteLine($"Person name changed: {newName}.");
        });

        person.Name = "Ilnaz"; // Prints a message about property change to the console

        // Disposing previously bound methods
        TriggerBinder.UnbindPropertyChanged(context: this, observable: person, p => p.Name);

        // Or, more generally:
        //TriggerBinder.UnbindPropertyChanged(context: this, observable: person);
        //TriggerBinder.UnbindPropertyChanged(context: this);
    }
}
``` 
I recently found out about your library, and it seems to me that it already has such mechanisms. Could you tell me if there is something similar and show an example? Thank you.

**Comment by ChrisPulman** [comment]
https://www.reactiveui.net/docs/handbook/data-binding/windows-presentation-foundation.html
This is a WPF example of how the bindings work

**Reply by Ilnazz** [reply]
Thank you for the answer. But is it possible using ReactiveUI to simply bind a lambda function to an object property change event, as in this piece of code:
```c#
TriggerBinder.OnPropertyChanged(context: this, observable: person, x => x.Name, newName =>
{
    Console.WriteLine($"Person name changed: {newName}.");
});
```
?

**Reply by Ilnazz** [reply]
Here is an example of how I use it in my own project:
```c#
using EasyBindings;
using ResearchProject.Models;

namespace ResearchProject.ViewModels;

public class ControlPanelViewModel : ViewModelBase
{
    public bool IsVisible
    {
        get => _panelStatesModel.IsControlPanelVisible;
        set => _panelStatesModel.IsControlPanelVisible = value;
    }

    // Has observable property IsControlPanelVisible
    private readonly PanelStatesModel _panelStatesModel;

    public ControlPanelViewModel(PanelStatesModel panelStatesModel)
    {
        _panelStatesModel = panelStatesModel;

        TriggerBinder.OnPropertyChanged(this, _panelStatesModel, o => o.IsControlPanelVisible, () =>
            OnPropertyChanged(nameof(IsVisible)));
    }
}
```
I decided to do this so that I would not create a separate _isVisible field in the view model, but directly access the property of the observable model; and when the property in the model changes, the view model will notify view about it.
And in general, is it right to do as I do? Maybe there's a better way?
